runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CombineTranslation.java

```diff
@@ -81,10 +73,14 @@ public String getUrn(Combine.PerKey<?, ?, ?> transform) {
     public FunctionSpec translate(
         AppliedPTransform<?, ?, Combine.PerKey<?, ?, ?>> transform, SdkComponents components)
         throws IOException {
-      return FunctionSpec.newBuilder()
-          .setUrn(COMBINE_TRANSFORM_URN)
-          .setPayload(payloadForCombine((AppliedPTransform) transform, components).toByteString())
-          .build();
+      if (transform.getTransform().getSideInputs().isEmpty()) {
+        return FunctionSpec.newBuilder()
+            .setUrn(COMBINE_TRANSFORM_URN)
+            .setPayload(payloadForCombine((AppliedPTransform) transform, components).toByteString())
+            .build();
+      } else {
+        return null;
```

Comment:
Add a comment here that generic composites have an empty/null FunctionSpec.

Reply:
Done.